src/lib/background/background.js

```diff
@@ -38,7 +46,11 @@ export class Background {
 
     this.scanQueue = new ScanQueue();
     this.scanQueue.bindScanComplete(this._handleScanComplete.bind(this));
+    this.scanQueue.bindQueueStateChange(
+      this._handleScanQueueStateChange.bind(this),
+    );
 
+    browser.browserAction.setIcon({path: defaultIcon});
```

Comment:
Semantically this should be done in _refreshIcon - keep all of the icon handling in one function.